src/components/reservation/reservation-modal.tsx

```diff
@@ -54,6 +54,7 @@ const ReservationModal = ({
   const [reservationResponse, setReservationResponse] =
     useState<ReservationResponseV2 | null>(null);
   const [selectedBranch, setSelectedBranch] = useState<string | null>(null);
+  const [selectedInterest, setSelectedInterest] = useState<string | null>(null);
```

Comment:
This is probably where we want to ensure that the interest day is a number not a string.